aten/src/THNN/generic/FusedRNNKernel.c

```diff
@@ -2,6 +2,31 @@
 #define TH_GENERIC_FILE "generic/FusedRNNKernel.c"
 #else
 
+void THNN_(FusedRNNAssertSizes)(int factor, int count, ...)
+{
+  va_list list;
+  va_start(list, count);
+
+  THTensor *input = va_arg(list, THTensor*);
+  THTensor *hidden = va_arg(list, THTensor*);
+  THArgCheck(THTensor_(nElement)(input) == THTensor_(nElement)(hidden),
+             3, "Input and Hidden tensor sizes should be the same.");
+
+  for (int arg = 2; arg < count; ++arg){
+    THTensor *tens = va_arg(list, THTensor*);
+    THArgCheck(THTensor_(nElement)(input) == THTensor_(nElement)(tens)*factor,
+               3, "A pointwise tensor was not the right size, should have 1/%u the elements of input/hidden tensor.", arg, factor);
+  }
+
+  va_end(list);
+}
+
+template <typename In, typename Out>
+inline Out THNN_(ScalarConvert)(In v) { return static_cast<Out>(v); }
+
+#define H2F(input) THNN_(ScalarConvert)<real, accreal>(input)
+#define F2H(input) THNN_(ScalarConvert)<accreal, real>(input)
```

Comment:
I don't think you can use C++ features in a C file. We might need to use a C++ compiler for that

Reply:
We do compile TH with C++. Maybe rename the file extension to `.cpp`?

Reply:
Is it worth thinking about doing this within ATen instead? FusedRNNKernel.c is very much empty.